File: src/Web/Front.Kashilog/ClientApp/src/features/Kashi/detail/types/company.ts
export class Company {
    name: string;
    zipCode: string;
    country: string;
    city: string;
    address: string;
    telephoneNumber: string;

    constructor(name: string, zipCode: string, country: string, city: string, address: string, telephoneNumber: string) {
        this.name = name;
        this.zipCode = zipCode;
        this.country = country;
        this.city = city;
        this.address = address;
        this.telephoneNumber = telephoneNumber;
    }

    public static validateAndGetAvailableValue(target: any): { validateResult: boolean, company: Company | null } {
        if (!(Object.hasOwn(target, 'companyName') &&
            Object.hasOwn(target, 'zipCode') &&
            Object.hasOwn(target, 'country') &&
            Object.hasOwn(target, 'city') &&
            Object.hasOwn(target, 'address') &&
            Object.hasOwn(target, 'telephoneNumber')
        )) {
            return { validateResult: false, company: null };
        }

        return {
            validateResult: true,
            company: new Company(
                target.companyName,
                target.zipCode,
                target.country,
                target.city,
                target.address,
                target.telephoneNumber
            )
        };
    }
}

File: src/Domain/DomainObject.Kashilog/Enterprise/Company.cs
namespace DomainObject.Kashilog.Enterprise {
    public class Company {

        public int CompanyId { get; set; }

        public string CompanyName { get; set; } = default!;

        public string ZipCode { get; set; } = default!;

        public string Country { get; set; } = default!;

        public string City { get; set; } = default!;

        public string Address { get; set; } = default!;

        public string TelephoneNumber { get; set; } = default!;
    }
}
